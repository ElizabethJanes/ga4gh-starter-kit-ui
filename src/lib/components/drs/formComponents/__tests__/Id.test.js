import React from 'react';
import renderer from 'react-test-renderer';
import Id from '../Id';

test('SHOW and EDIT <Id /> should handle empty id field', () => {
    const id = renderer.create(<Id readOnly={true} id='' />)
    .toJSON();
    expect(id).toMatchSnapshot();
});

test('SHOW and EDIT <Id /> should handle populated id field', () => {
    const id = renderer.create(<Id readOnly={true} id='1234567890' />)
    .toJSON();
    expect(id).toMatchSnapshot();
});

test('NEW <Id /> should handle empty id field', () => {
    const id = renderer.create(<Id readOnly={false} id='' />)
    .toJSON();
    expect(id).toMatchSnapshot();
});

test('NEW <Id /> should handle populated id field', () => {
    const id = renderer.create(<Id readOnly={false} id='1234567890' />)
    .toJSON();
    expect(id).toMatchSnapshot();
});