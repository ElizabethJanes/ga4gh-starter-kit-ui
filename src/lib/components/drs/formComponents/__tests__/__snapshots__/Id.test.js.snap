// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW <Id /> should handle empty id field 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="id"
          id="id-label"
        >
          Id
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="id-helper-text"
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="id"
            name="id"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
        </div>
        <p
          className="MuiFormHelperText-root"
          id="id-helper-text"
        >
          Unique identifier for this DRS Object (UUID recommended), cannot be modified later.
        </p>
      </div>
    </div>
  </div>
  <div
    align="right"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiTypography-root MuiTypography-button"
        >
          Generate ID
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`NEW <Id /> should handle populated id field 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="id"
          id="id-label"
        >
          Id
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="id-helper-text"
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="id"
            name="id"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={false}
            required={false}
            type="text"
            value="1234567890"
          />
        </div>
        <p
          className="MuiFormHelperText-root MuiFormHelperText-filled"
          id="id-helper-text"
        >
          Unique identifier for this DRS Object (UUID recommended), cannot be modified later.
        </p>
      </div>
    </div>
  </div>
  <div
    align="right"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiTypography-root MuiTypography-button"
        >
          Generate ID
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`SHOW and EDIT <Id /> should handle empty id field 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="id"
          id="id-label"
        >
          Id
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="id-helper-text"
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="id"
            name="id"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value=""
          />
        </div>
        <p
          className="MuiFormHelperText-root"
          id="id-helper-text"
        >
          Unique identifier for this DRS Object (UUID recommended), cannot be modified later.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`SHOW and EDIT <Id /> should handle populated id field 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
          htmlFor="id"
          id="id-label"
        >
          Id
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="id-helper-text"
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="id"
            name="id"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="1234567890"
          />
        </div>
        <p
          className="MuiFormHelperText-root MuiFormHelperText-filled"
          id="id-helper-text"
        >
          Unique identifier for this DRS Object (UUID recommended), cannot be modified later.
        </p>
      </div>
    </div>
  </div>
</div>
`;
